code/modules/assembly/flash.dm

```diff
@@ -322,6 +323,62 @@
 /obj/item/assembly/flash/cyborg/screwdriver_act(mob/living/user, obj/item/I)
 	return
 
+///Camera flash
+
+/obj/item/assembly/flash/camera
+	name = "camera"
+	icon = 'icons/obj/items_and_weapons.dmi'
+	desc = "A polaroid camera."
+	icon_state = "camera"
+	inhand_icon_state = "camera"
+	lefthand_file = 'icons/mob/inhands/misc/devices_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/misc/devices_righthand.dmi'
+	slot_flags = ITEM_SLOT_NECK
+	pickup_sound = null //override pickup and drop sounds so it behaves more like a real camera.
+	drop_sound = null
+	flash_sound = 'sound/items/polaroid1.ogg'
+	var/max_charges = 5
+	var/current_charges = 5
+	var/list/charge_timers = list()
+	var/charge_time = 30 SECONDS
+
+/obj/item/assembly/flash/camera/burn_out()
+	return //use self charging system instead
+
+/obj/item/assembly/flash/camera/attack_self(mob/living/carbon/user, flag = 0)
+	if(current_charges)
+		current_charges--
+		to_chat(user, span_notice("You use [src]. It now has [current_charges] charge\s remaining."))
+		charge_timers.Add(addtimer(CALLBACK(src, .proc/recharge), charge_time, TIMER_STOPPABLE))
+		..()
+	else
+		to_chat(user, span_warning("[src] is recharging."))
+
+/obj/item/assembly/flash/camera/attack(mob/living/M, mob/user)
+	if(current_charges)
+		current_charges--
+		to_chat(user, span_notice("You use [src]. It now has [current_charges] charge\s remaining."))
+		charge_timers.Add(addtimer(CALLBACK(src, .proc/recharge), charge_time, TIMER_STOPPABLE))
+		..()
+	else
+		to_chat(user, span_warning("[src] is recharging."))
+
+/obj/item/assembly/flash/camera/proc/recharge(mob/user)
```

Comment:
Since both of these procs have the same behaviour, it'd be better to move that out into its own separate proc and call that on both rather than having it copy/pasted. E.g:
```suggestion
/obj/item/assembly/flash/camera/attack_self(mob/living/carbon/user, flag = 0)
	if(use_charge(user))
		. = ..()

/obj/item/assembly/flash/camera/attack(mob/living/M, mob/user)
	if(use_charge(user))
		. = ..()

/obj/item/assembly/flash/camera/proc/use_charge(mob/user)
	if(!current_charges)
		to_chat(user, span_warning("[src] is recharging."))
		return FALSE

	current_charges--
	to_chat(user, span_notice("You use [src]. It now has [current_charges] charge\s remaining."))
	charge_timers.Add(addtimer(CALLBACK(src, .proc/recharge), charge_time, TIMER_STOPPABLE))
	return TRUE

/obj/item/assembly/flash/camera/proc/recharge(mob/user)
```